### PR TITLE
feat: allow customize cgroup root path

### PR DESCRIPTION
**Pull request by fengxsong**
https://github.com/containers/youki/issues/2392

### CODE REVIEW

**Comment by fengxsong** [comment]
@utam0k PTAL

**Comment by utam0k** [comment]
@fengxsong May I ask you to make it ready?

File: crates/libcgroups/src/common.rs [hunk 1]
```diff
@@ -335,10 +339,17 @@ pub struct CgroupConfig {
...
+pub fn create_cgroup_manager_with_root(
+    root_path: &Path,
     config: CgroupConfig,
 ) -> Result<AnyCgroupManager, CreateCgroupSetupError> {
-    let cgroup_setup = get_cgroup_setup().map_err(|err| match err {
+    let root = if root_path.is_empty() {
+        Path::new(DEFAULT_CGROUP_ROOT)
+    } else {
+        root_path
+    };
```
Comment (utam0k) [inline crates/libcgroups/src/common.rs:350]: How about using `Option`?
https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap_or_default

**Review by utam0k — CHANGES_REQUESTED** [review]
Thanks for your PR. I left a comment.

File: crates/libcgroups/src/common.rs (hunk 1 above)
Comment (fengxsong) [reply crates/libcgroups/src/common.rs]: good idea. I've changed it.

File: crates/libcgroups/src/common.rs [hunk 2]
```diff
@@ -335,10 +338,16 @@ pub struct CgroupConfig {
     pub container_name: String,
 }
 
-pub fn create_cgroup_manager(
+pub fn create_cgroup_manager_with_root(
```
Comment (utam0k) [inline crates/libcgroups/src/common.rs:341]: Please comment on the behavior of passing None

Comment (fengxsong) [reply crates/libcgroups/src/common.rs:343]: done

**Review by utam0k — APPROVED** [review]
Thanks for your contribution 😍

**Pull request merged**